apps/client/vite.config.ts

```diff
@@ -0,0 +1,53 @@
+import { defineConfig } from 'vite';
+import react from '@vitejs/plugin-react';
+import dynamicImport from 'vite-plugin-dynamic-import';
+import commonjs from 'vite-plugin-commonjs';
+import path from 'path';
+
+export default defineConfig(() => {
+  return {
+    build: {
+      outDir: 'build',
+    },
+    plugins: [
```

Comment:
need to configure `app-kit` to be a singleton to utilize the metrics [current config](https://github.com/awslabs/iot-application/commit/2a8208950306351f04f550ec237a5b73e7231c7a#diff-a442e6ce9c8ed6d5ecaf9f47bff7a20278cff7d19f56c148f4a4f32bac5ea4a5)
I will help looking for the vite counterpart as well 🙏

Reply:
please try the config here https://github.com/originjs/vite-plugin-federation?tab=readme-ov-file#packagepath-string
with shared of
```
      shared: ['@iot-app-kit/core'],
```

Reply:
As we discussed offline, I added 

```
  federation({
    shared: ['@iot-app-kit/core']
  }),
```

to the plugins which will create the singleton instance as expected. I tested in the cloud deployment and saw the metrics being sent.

I also had to add a plugin to remove the topLevelAwaits from core.

Reply:
As we discussed offline, we're going to hold off on adding this plugin and instead find an alternative for creating a singleton app kit instance for our logging solution. With this config, we're adding top level awaits, then having to add another plugin to remove them, and it's breaking our tests. We will follow up with a more stable solution.